fix/add-mood-keyword-to-match-api-service

Mostafa-khaled21 opened the pull request:
add keyword to Api service
localization for duration and season count
fixed  the error when I chose both in release 

https://github.com/user-attachments/assets/4f7add1b-d295-442c-bd8b-eca4b098cd3d

Mostafa-khaled21 committed:
feat: add moodId parameter to GetMatchingMoviesUseCase and GetMatchingSeriesUseCase

This commit introduces the `moodId` parameter to `GetMatchingMoviesUseCase` and `GetMatchingSeriesUseCase` and their respective tests. This allows filtering movies and series by mood.

Mostafa-khaled21 committed:
feat: add moodId parameter to match media

This commit adds a `moodId` parameter to the `getMatchingMovies` and `getMatchingSeries` functions in the `MatchApiService`, `MatchRemoteDataSource`, `MatchRepository`, and their implementations. This allows filtering media by mood.

Mostafa-khaled21 committed:
refactor: improve error handling in RetrofitRequestBuilder

The `safeCall` function now catches generic `Throwable` exceptions and maps them to `MediaDataException` for more consistent error handling.

Mostafa-khaled21 committed:
feat: implement localization for date, duration, and season count

Mostafa-khaled21 committed:
Fix empty state and filter by mood in match result screen

This commit introduces the following changes:
- Displays an empty state message when no match results are found.
- Adds mood filtering capability to the match result screen.
- Updates the `HeroCarousel` initial page to be 0 instead of 2.
- Refactors string formatting for duration and release date.

Mostafa-khaled21 committed:
feat: add new strings for empty state and season count

Add new strings for:
- No results title
- No results caption
- Try again button
- Season count with singular/plural support

Update Arabic translations for the new strings.

mohamedebrahem13 commented on presentation/match/src/main/java/com/giraffe/match/utils/formatAsFullDate.kt:
```diff
@@ -1,21 +1,38 @@
 
+
+private val arabicDigits = listOf('٠', '١', '٢', '٣', '٤', '٥', '٦', '٧', '٨', '٩')
+private val arabicMonths = listOf(
```
doesn't dart format support Arabic character?

mohamedebrahem13 commented on presentation/match/src/main/java/com/giraffe/match/utils/formatAsFullDate.kt:
```diff
@@ -1,21 +1,38 @@
-    } else {
-        "%04d-%02d-%02d".format(this.year, this.month.number, this.day)
-    }
+    val locale = Locale.getDefault()
```
We don't use local now to get the language

Mostafa-khaled21 replied:
I want to know the language because the date appears in English even if the phone is in Arabic, So I translated it

HendSayed25 commented on datasource/remote/media/src/main/java/com/giraffe/media/match/retrofit/MatchRemoteDataSourceImpl.kt:
```diff
@@ -15,14 +15,16 @@ class MatchRemoteDataSourceImplRetrofit @Inject constructor(
         maxRuntime: Int?,
         earliestFirstAirDate: String?,
-        latestFirstAirDate: String?
+        latestFirstAirDate: String?,
```
Air in latestFirstAirDate refer to what ?

Mostafa-khaled21 committed:
refactor: rename date parameters in match module

The query parameters `earliestFirstAirDate` and `latestFirstAirDate` have been renamed to `earliestDate` and `latestDate` respectively in `MatchApiService.kt`, `MatchRemoteDataSourceImpl.kt`, and `MatchRemoteDataSource.kt`.

Mostafa-khaled21 committed:
Refactor date formatting for localization

Refactored the `formatAsFullDate` function to improve localization support.

- Removed hardcoded Arabic and English month names and Arabic digit mappings.
- Utilized `AppCompatDelegate.getApplicationLocales()` for Android Tiramisu (API 33) and above to get the primary application locale.
- Used `java.time.Month.getDisplayName(TextStyle.FULL, locale)` for Android Oreo (API 26) and above to get localized month names.
- For versions below Android Oreo, the month number is used as a fallback.
- Arabic digit conversion is now done using a string lookup for conciseness.

Mostafa-khaled21 committed:
Refactor: Simplify locale retrieval in `formatAsFullDate`

The `if/else` block for Android versions was removed, simplifying the locale retrieval to directly use `AppCompatDelegate.getApplicationLocales().get(0)` as the behavior is consistent across versions.
